api/src/main/java/org/openmrs/module/registrationcore/api/mpi/openempi/OpenEmpiImplementation.java

```diff
@@ -0,0 +1,78 @@
+package org.openmrs.module.registrationcore.api.mpi.openempi;
+
+import org.openmrs.Patient;
+import org.openmrs.module.registrationcore.api.mpi.common.*;
+import org.openmrs.module.registrationcore.api.search.PatientAndMatchQuality;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.util.List;
+import java.util.Map;
+
+public class OpenEmpiImplementation implements MpiProvider {
+
+    @Autowired
+    @Qualifier("registrationcore.mpiPatientImporter")
+    private MpiPatientImporter patientImporter;
```

Comment:
So, I dislike this: the OpenEMPI-specific implementation should have OpenEMPI-specific things wired into it.

To put it another way, I would expect to have one (bigger) class with a bunch of OpenEMPI-specific implementations, rather than having class-per-function which are supposed to handle both OpenEMPI and other MPIs.